libs/core/src/components/spinner/gds-spinner.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * @fileoverview GDS Spinner Component - A loading indicator with accessibility support
+ * @module gds-spinner
+ */
+
+import { css } from 'lit'
+import { property } from 'lit/decorators.js'
+
+import { GdsElement } from '../../gds-element'
+import { tokens } from '../../tokens.style'
+import { TransitionalStyles } from '../../transitional-styles'
+import {
+  gdsCustomElement,
+  html,
+} from '../../utils/helpers/custom-element-scoping'
+import { styles } from './gds-spinner.styles'
+
+/**
+ * GdsSpinner Component
+ * A loading indicator web component that supports various sizes and display modes.
+ *
+ * @extends {GdsElement}
+ *
+ * @example
+ * ```html
+ * <!-- Basic usage -->
+ * <gds-spinner size="md" text="Loading..."></gds-spinner>
+ *
+ * <!-- Screen reader only text -->
+ * <gds-spinner size="md" text="Loading..." .visualText=${false}></gds-spinner>
+ *
+ * <!-- Cover container -->
+ * <gds-spinner size="md" text="Loading..." cover></gds-spinner>
+ * ```
+ *
+ * @csspart spinner - The spinning circle element
+ * @csspart text - The loading text element
+ *
+ * @fires gds-element-disconnected - Fired when the element is removed from the DOM
+ */
+@gdsCustomElement('gds-spinner')
+export class GdsSpinner extends GdsElement {
+  /**
+   * Component styles combining design tokens, host styles, and component-specific styles
+   * @type {Array<CSSStyleSheet>}
+   */
+  static styles = [
+    tokens,
+    css`
+      :host {
+        display: block;
+      }
+    `,
+    styles,
+  ]
+
+  /**
+   * Loading text to be displayed. Can be shown visually or for screen readers only.
+   * @type {string|undefined}
+   */
+  @property({ type: String })
+  text?: string
+
+  /**
+   * Controls text visibility mode.
+   * When true, text is visible and announced to screen readers.
+   * When false, text is only announced to screen readers.
+   * @type {boolean}
+   * @default true
+   */
+  @property({ type: Boolean })
+  visualText: boolean = true
+
+  /**
+   * When true, adds a semi-transparent backdrop over the container.
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean })
+  cover: boolean = false
+
+  /**
+   * When true, spinner covers the entire viewport with backdrop.
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean })
+  fullscreen: boolean = false
+
+  /**
+   * Controls the size of the spinner.
+   * @type {'default'|'sm'|'md'|'lg'}
+   * @default 'default'
+   */
+  @property({ type: String })
+  size: 'default' | 'md' | 'lg' | 'sm' = 'default'
+
+  /**
+   * Lifecycle method called when component is added to DOM.
+   * Sets up accessibility attributes and applies transition styles.
+   * @return {void}
+   */
+  connectedCallback(): void {
+    super.connectedCallback()
+    this.setAttribute('role', 'status')
```

Comment:
In terms of semantics I think 'progressbar' is a better suited role. Then we can also extend it in the future to convey quantifiable progress. Also, `aria-busy` should be added to the region which will be updated when loading is done. To keep it simple, I think we'll need to leave that to the consumer for now.

Here is an example of indeterminate progress using the `<progress>` element:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress#describing_a_particular_region

Maybe down the road we can add a parent component, something like `<gds-progress>`, that can wrap a loading region to add a spinner or progress bar and automatically wire everything up with correct aria roles.

Reply:
Status vs. Progressbar for Spinners
Using role="status":
Perfectly valid for spinners that don't show measurable progress
Simpler implementation with fewer required attributes
Communicates to screen readers: "Here's a status update about loading"
Many design systems and accessibility experts recommend this for indeterminate spinners
Using role="progressbar" with undefined aria-valuenow:
More semantically tied to the concept of "progress in process"
Requires additional attributes (aria-valuemin, aria-valuemax)
Some screen readers may announce it differently (as a progress indicator)
Better suited if your spinner might eventually display determinate progress
The WAI-ARIA authoring practices actually support both approaches for indeterminate spinners.

Reply:
Allright, I buy this argument. Let's leave it!